src/lib/y2storage/device.rb

```diff
@@ -0,0 +1,98 @@
+require "y2storage/storage_class_wrapper"
+
+module Y2Storage
+  # An abstract base class of storage devices and a vertex in the Devicegraph.
+  #
+  # The Device class does not have a device name since some device types do
+  # not have a intrinsic device name, e.g. btrfs.
+  #
+  # This is a wrapper for Storage::Device
+  class Device
+    include StorageClassWrapper
+    wrap_class Storage::Device,
+      downcast_to: ["BlkDevice", "Mountable", "PartitionTables::Base"]
+
+    storage_forward :==
+    storage_forward :!=
+    storage_forward :sid
+    storage_forward :ancestors, as: "Device"
+    storage_forward :exists_in_devicegraph?
+    storage_forward :exists_in_probed?
+    storage_forward :exists_in_staging?
+    storage_forward :displayname
+    storage_forward :detect_resize_info
+    storage_forward :remove_descendants
+    storage_forward :userdata
+    storage_forward :userdata=
+
+    # Checks where the device is a disk
+    #
+    # @return [Boolean] true if this is a Y2Storage::Disk
+    def disk?
```

Comment:
I added them for convenience during queries. To shorten relatively common checks like:
```ruby
a = filesystem.blk_devices.select(&:disk?)
b= another_filesystem.ancestors.select { |a| a.disk? || a.partition? }
```
That's much readable than
```ruby
a = filesystem.blk_devices.select { |a| a.is_a?(Y2Storage::Disk) }
b= another_filesystem.ancestors.select { |a| is_a?(Y2Storage::Disk) || is_a?(Y2Storage::Partition) }
```